xonsh/xontribs_meta.py

```diff
@@ -0,0 +1,639 @@
+import functools
+import typing as tp
+
+
+class XontribPkg(tp.NamedTuple):
+    install: tp.Dict[str, str]
+    license: str = ""
+    name: str = ""
+    url: tp.Optional[str] = None
+
+
+class Xontrib(tp.NamedTuple):
+    """Meta class that is used to descript xontribs.
+
+    Attributes
+    ----------
+        url: str
+            url
+        description: str
+            short description about the xontrib.
+        package: XontribPkg
+            pkg information for installing the xontrib
+        tags: tuple of str
+            category.
+    """
+
+    url: str = ""
+    description: str = ""
+    package: tp.Optional[XontribPkg] = None
+    tags: tp.Tuple[str, ...] = ()
+
+
+@functools.lru_cache()
+def get_xontribs() -> tp.Dict[str, Xontrib]:
```

Comment:
get doesn't seem like the right verb here.  find or make, maybe? Just a suggestion

Reply:
define_xontribs maybe ?

Reply:
the thing is, it is also a getter from other modules.

Reply:
Yeah that is fair

Reply:
sorry not clear, whether you prefer define_xontrib or keep get_xontribs

Reply:
It is your PR so you can choose the name 😉

Reply:
well then we will go with get_xontribs for now, in case we split the definitions to a new file or one we can use TypedDict, we can ditch requiring to pass the class name altogether.